src/ILCompiler/src/Program.cs

```diff
@@ -170,6 +188,51 @@ private int Run(string[] args)
             return 0;
         }
 
+        private TypeDesc FindType(CompilerTypeSystemContext context, string typeName)
+        {
+            TypeDesc foundType = context.SystemModule.GetTypeByCustomAttributeTypeName(typeName);
+            if (foundType == null)
+            {
+                throw new CommandLineException($"Type '{typeName}' not found");
+            }
+            return foundType;
+        }
+
+        private MethodDesc CheckAndParseSingleMethodModeArguments(CompilerTypeSystemContext context)
+        {
+            if (_singleMethodName == null && _singleMethodTypeName == null && _singleMethodGenericArgs == null)
+                return null;
+
+            if (_singleMethodName == null || _singleMethodTypeName == null)
+                throw new CommandLineException("Both method name and type name are required parameters for single method mode");
+
+            TypeDesc owningType = FindType(context, _singleMethodTypeName);
```

Comment:
Yup. The type name is in the Ecma-335 SerString format. It can express anything a TypeOf can express, including generic types instantiated over other generic types over arrays, etc.